File: java/placedOrdersStatistics/src/entities/Item.java
package entities;

public class Item {
    private Product product;
    private Double cost;
    private Double shippingFee;
    private Double taxAmount;

    public Item(Product product, Double shippingFee, Double taxAmount) {
        this.product = product;
        this.shippingFee = shippingFee;
        this.taxAmount = taxAmount;
        this.cost = shippingFee + taxAmount + product.getPrice();
    }

    public Product getProduct() {
        return product;
    }

    public void setProduct(Product product) {
        this.product = product;
    }

    public Double getCost() {
        return cost;
    }

    public void setCost(Double cost) {
        this.cost = cost;
    }

    public Double getShippingFee() {
        return shippingFee;
    }

    public void setShippingFee(Double shippingFee) {
        this.shippingFee = shippingFee;
    }

    public Double getTaxAmount() {
        return taxAmount;
    }

    public void setTaxAmount(Double taxAmount) {
        this.taxAmount = taxAmount;
    }

    @Override
    public String toString() {
        return "Item{" +
                "product=" + product +
                ", cost=" + cost +
                ", shippingFee=" + shippingFee +
                ", taxAmount=" + taxAmount +
                '}';
    }
}

File: scala/src/main/scala/entity/Item.scala
package entity

class Item (
           //private var id: Long,
           private var product: Product,
           private var shippingFee: Double,
           private var taxAmount: Double
           ){
  private val cost: Double = shippingFee + taxAmount + product.getPrice

  //def getId: Long = id

  def getProduct: Product = product

  def setProduct(newProduct: Product): Unit = {
    product = newProduct
  }

  def getShippingFee: Double = shippingFee

  def setShippingFee(newShippingFee: Double): Unit = {
    shippingFee = newShippingFee
  }

  def getTaxAmount: Double = taxAmount

  def setTaxAmount(newTaxAmount: Double): Unit = {
    taxAmount = newTaxAmount
  }

  def getCost: Double = cost


  override def toString: String = {
    s"Produto: ${product.toString}, Custo: $cost, Taxa de Envio: $shippingFee, Valor do Imposto: $taxAmount"
  }

}
